dgp/lib/plots.py

```diff
@@ -128,3 +127,52 @@ def mapplot_line(pnt_full, pnt, data, var, units='', ptitle='test_map', pfile='t
     except IndexError:
         print("Couldn't make Map Plot.")
     return
+
+
+def mapplot_segment(df, var, begin_segment, end_segment, units='', ptitle='test_map', pfile='test_map'):
+    """
+    This makes a map plot of line segment
+    :param df:
+    :param var:
+    :param units:
+    :param ptitle:
+    :param pfile:
+    :return:
+    """
+    try:
+        data = df[var][(begin_segment <= df[var].index) & (df[var].index <= end_segment)]
+        lat = df['trajectory'].lat[(begin_segment <= df['trajectory'].lat.index)
+                                   & (df['trajectory'].lat.index <= end_segment)]
+        long = df['trajectory'].long[(begin_segment <= df['trajectory'].long.index)
+                                    & (df['trajectory'].long.index <= end_segment)]
+        box = sgeom.box(minx=long.min() - 1, maxx=long.max() + 1,
+                        miny=lat.min() - 1, maxy=lat.max() + 1)
+        x0, y0, x1, y1 = box.bounds
+        if x0 < 0:
+            myproj = ccrs.SouthPolarStereo(central_longitude=180)
+        else:
+            myproj = ccrs.NorthPolarStereo(central_longitude=0)
+        fig = plt.figure(figsize=(8, 4), facecolor='white', dpi=144)
+        ax = plt.axes(projection=myproj)
+
+        s2 = plt.scatter(long, lat, c=data, cmap=cm.Spectral, s=10, transform=ccrs.PlateCarree())
+        cb = fig.colorbar(s2, ax=ax, label=units,
+                          orientation='vertical', shrink=0.8, pad=0.05)
+        cb.ax.set_yticklabels(cb.ax.get_yticklabels(), rotation=0)
+        cb.ax.tick_params(labelsize=6)
+
+        ax.coastlines(resolution='10m')
+        ax.xlabels_top = ax.ylabels_right = False
+        ax.gridlines(draw_labels=False, alpha=0.3, color='grey')
+        ax.xformatter = LONGITUDE_FORMATTER
+        ax.yformatter = LATITUDE_FORMATTER
+        # ax.set_extent([x0, x1, y0, y1], ccrs.PlateCarree())
+
+        plt.tight_layout()
+        plt.subplots_adjust(top=0.90)
+        plt.suptitle(ptitle, y=0.98)
+        plt.savefig(pfile, bbox_inches='tight')  # save the figure to file
+        plt.close()
+    except IndexError:
```

Comment:
You should aim to put the minimum required lines in between a try-except.  Which line or lines in between try and except can possibly throw an `IndexError`?  And then make the error more specific.